overlord/state/task.go

```diff
@@ -163,13 +167,48 @@ func (t *Task) SetProgress(cur, total int) {
 	t.progress = progress{Current: cur, Total: total}
 }
 
+const (
+	// Messages logged in tasks are guaranteed to use the following strings
+	// plus ": " as a prefix, so these may be handled programatically and
+	// stripped for presentation.
+	LogInfo  = "INFO"
+	LogError = "ERROR"
+)
+
+func (t *Task) logf(kind, format string, args ...interface{}) {
+	if len(t.log) > 9 {
```

Comment:
So we-log rotate at 10 items? I don't really mind it here but there are more efficient ways to keep a circular log.